pkg/snapshot/snapshotter/garbagecollector.go

```diff
@@ -140,6 +142,10 @@ func (ssr *Snapshotter) RunGarbageCollector(stopCh <-chan struct{}) {
 					}
 
 					if deleteSnap {
+						if !nextSnap.IsDeletable() {
+							ssr.logger.Infof("GC: Skipping : %s, since it is immutable", nextSnap.SnapName)
```

Comment:
```suggestion
							ssr.logger.Infof("GC: Skipping the snapshot: %s, since it's retention period not expired", nextSnap.SnapName)
```

Reply:
Addressed in [6b5b125](https://github.com/gardener/etcd-backup-restore/pull/776/commits/6b5b125f5bbb9044539f29992e62e52ac7a0bfa1).